libcontainer/sync.go

```diff
@@ -47,7 +53,13 @@ func (i initError) Error() string {
 // writeSync is used to write to a synchronisation pipe. An error is returned
 // if there was a problem writing the payload.
 func writeSync(pipe io.Writer, sync syncType) error {
-	return utils.WriteJSON(pipe, syncT{sync})
+	return utils.WriteJSON(pipe, syncT{sync, -1})
+}
+
+// writeSyncWithFd is used to write to a synchronisation pipe. An error is
+// returned if there was a problem writing the payload.
+func writeSyncWithFd(pipe io.Writer, sync syncType, fd int) error {
+	return utils.WriteJSON(pipe, syncT{sync, fd})
```

Comment:
So,

1. `writeSync` can call `writeSyncWithFd(pipe, sync, -1)`.

2. `writeSyncWithFd` can wrap the error from `WriteJSON` so it's clear as to what is going on, e.g.

```go
func writeSyncWithFd(pipe io.Writer, sync syncType, fd int) error {
	err := utils.WriteJSON(pipe, syncT{sync, fd})
	if err == nil {
		return nil
	}
	return fmt.Errorf("writing syncT %q: %w", string(sync), err)
}
```

Reply:
Ohh, sorry, I didn't get that. I've changed it to that and made sure that all callers just return the error (only paths in this PR needed to be adjusted to avoid errors like: `writing syncT SeccompDone: writing syncT SeccompDone: xx` and have just `writing syncT SeccompDone: xx`)

Thanks!